src/Control/Monad/Catch.hs

```diff
@@ -6,6 +6,7 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE TypeFamilies #-}
```

Comment:
Out of curiosity, would this module compile with https://gitlab.haskell.org/ghc/ghc/-/merge_requests/5597 if we removed the use of `TypeFamilies` here? If so, it would be worth including a comment next to this line stating that `TypeFamilies` is only enabled for backwards compatibility with older GHCs. In the distant future, we might even be able to remove the use of `TypeFamilies` here altogether.

Reply:
I wanted to write

```
{-# LANGUAGE TypeFamilies #-}    -- needed to use type equality with GHC <= 9.4
```

But we don't know yet if the feature will make it into 9.4. What about opening an issue to track the removal of `TypeFamilies` instead of adding a comment?

Reply:
Thanks for checking. No need to file an issue about this—I was mostly asking out of curiosity. Since it's uncertain when this feature will debut, I'm content with just adding another language extension. If I really care about pruning unnecessary language extensions, I should probably use `hlint` here :)